### PR TITLE
Add Icon Button Component

### LINKED ISSUE #46: Implement Icon Button Component
- [x] Research
- [ ] Create the specification
- [ ] Implement the component and tests
- [x] API Documentation
- [ ] Design implementation

### PR DESCRIPTION
**Pull request by simeonoff**
Closes #46

### CODE REVIEW

File: src/components/button/icon-button.indigo.scss [hunk 1]
```diff
@@ -0,0 +1,77 @@
+@use '../../styles/utilities' as utils;
+
+@mixin theme() {
+    igc-icon-button::part(base) {
+        transition: color .15s ease-in-out,
+            box-shadow .15s ease-in-out,
```
Comment (SisIvanova) [inline src/components/button/icon-button.indigo.scss:6]:
```suggestion
           box-shadow .15s ease-in-out,
```

File: src/components/button/icon-button.indigo.scss [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            background .15s ease-in-out;
```
Comment (SisIvanova) [inline src/components/button/icon-button.indigo.scss:7]:
```suggestion
           background .15s ease-in-out;
```

File: src/components/button/icon-button.material.scss [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+:host {
+    display: inline-flex;
+}
+
+[part='base'] {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    border: none;
+    outline-style : none;
```
Comment (SisIvanova) [inline src/components/button/icon-button.material.scss:13]:
```suggestion
    outline-style: none;
```

File: src/components/button/icon-button.spec.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+      name="biking"
+      collection="default"
+      variant="contained"
+      size="medium"
+      href="https://test.com"
+      target="_blank"
+      download="test"
+      rel="test">
+      Submit
+      <igc-link-button>`
```
Comment (SisIvanova) [inline src/components/button/icon-button.spec.ts:173]:
```suggestion
      </igc-icon-button>`
```

File: stories/card.stories.ts [hunk 5]
```diff
@@ -57,7 +57,7 @@ const Template: Story<ArgTypes, Context> = (
         <igc-card-actions>
           <igc-button slot="start" variant="flat">Like</igc-button>
           <igc-button slot="start" variant="flat">Learn More</igc-button>
-          <igc-icon slot="end" name="home" collection="default"></igc-icon>
+          <igc-icon-button slot="end" name="home"></igc-icon>
```
Comment (SisIvanova) [inline stories/card.stories.ts:60]:
```suggestion
          <igc-icon-button slot="end" name="home"></igc-icon-button>
```

**Review by SisIvanova — DISMISSED** [review]

File: src/components/button/icon-button.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+  /** The target attribute of the icon. */
+  @property()
+  target!: '_blank' | '_parent' | '_self' | '_top' | undefined;
+
+  /** The rel attribute of the icon. */
+  @property()
+  rel!: string;
+
+  protected renderIcon() {
```
Comment (gedinakova) [inline src/components/button/icon-button.ts:48]: `private` instead of `protected`?

File: src/components/button/icon-button.ts [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+  /** The name of the icon collection. */
+  @property()
+  collection!: string;
+
+  /** Whether to flip the icon. Useful for RTL layouts. */
+  @property({ type: Boolean })
+  mirrored = false;
+
+  /** The href attribute of the icon. */
```
Comment (gedinakova) [inline src/components/button/icon-button.ts:28]: From here on, the comments should probably mention `icon button` instead of just `icon`.

**Review by gedinakova — CHANGES_REQUESTED** [review]
Please, add the appropriate access modifiers to all API members.

File: src/components/button/icon-button.ts [hunk 8]
```diff
@@ -0,0 +1,92 @@
+import { html } from 'lit';
```
Comment (gedinakova) [inline src/components/button/icon-button.ts:1]: Add access modifiers to all API members.

**Review by gedinakova — CHANGES_REQUESTED** [review]

**Review by gedinakova — APPROVED** [review]

**Pull request merged**